### PR TITLE
Renaming CPU architecture to have consistent naming

### LINKED ISSUE #606: [BUG] CPU Architectures naming is not consistent with OpenSearch plugins
**Describe the bug**
CPU architectures are named `x86_64` and `aarch64` in OpenSearch, where as rest of the plugins use `x64` and `arm64`.
Also platform for MacOS has different naming conventions, OpenSearch/Dashboards use `darwin` and the plugins use `macos`.

- [ ] CPU architecture - Done PR #612 
- [ ] MacOS Platform

Also the google trends show the latter are on the rise.

### PR DESCRIPTION
**Pull request by saratvemulapalli**
### Description
Renaming CPU architecture to have consistent naming.
Here are the following changes:
1. x86_64 to x64
2. aarch64 to arm64
 
### Issues Resolved
#606 
 
### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by saratvemulapalli** [comment]
start gradle check

**Comment by saratvemulapalli** [comment]
start gradle check

File: buildSrc/src/main/java/org/opensearch/gradle/Jdk.java [hunk 1]
```diff
@@ -229,4 +230,15 @@ private void parseVersion(String version) {
         hash = jdkVersionMatcher.group(5);
     }
 
+    private String translateJdkArchitecture(String architecture) {
```
Comment (harold-wang) [inline buildSrc/src/main/java/org/opensearch/gradle/Jdk.java:233]: Do u want to change to return architecture == "arm64"? "aarch64" : architecture; ? thx

Comment (saratvemulapalli) [reply buildSrc/src/main/java/org/opensearch/gradle/Jdk.java:233]: I like that, sure.

**Review by dblock — APPROVED** [review]

**Comment by saratvemulapalli** [comment]
start gradle check

**Review by harold-wang — APPROVED** [review]
Looks good to me, thanks for the change!

**Pull request merged**